### PR TITLE
[REG-1636] Replays state recordings using the new unity input system only

### PR DESCRIPTION
**Pull request by RG-nAmKcAz**
- Replays state recordings using the new unity input system only (any code using Input.Get.. will NOT trigger)
- Fixes an annoyance where we log 2 errors when your RG credentials aren't setup
- Changes the state recording code and file format since the previous PR.. No, it will not be backwards compatible at this point, just make a new recording for your testing.
- At this time, this is fairly stable all things considered, but does NOT support hover over interactions very well.
- Includes 3rd party code for runtime SimpleFileBrowser (https://github.com/yasirkula/UnitySimpleFileBrowser)
- Adds 2 new behaviour scripts, 1 for excluding an object and its children from the state (used for RGOverlay), 1 for including an object in the state... useful for empty parent gameobjects that you may want state to roll up under

Demo Video: 

https://github.com/Regression-Games/RGUnityBots/assets/112959031/dae37f13-81ba-4f25-85e8-edeb3ab6029a

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

**Review by abeizer — APPROVED** [review]
The demo video is neat, I'm excited to consume this work for validations. I see a few TODOs that may need linear IDs, but I don't have any blocking comments. Just general coolness.

Side-note: Never have I imagined a file picker being this large. Wow.

**Comment by RG-nAmKcAz** [comment]
@abeizer Re: File Picker... free runtime file picker asset so I didn't have to do the work... we can enhance the appearance later

**Comment by abeizer** [comment]
@RG-nAmKcAz yes, I saw. That wasn't a criticism of your PR - just funny ;-)

File: src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ReplayDataPlaybackController.cs [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+            }
+        }
+
+        private InputDevice GetMouse(bool forceListener = false)
+        {
+            var mouse = InputSystem.devices.FirstOrDefault(a => a.name == "RGVirtualMouse");
+            var created = false;
+            if (mouse == null)
+            {
+                mouse = InputSystem.AddDevice<Mouse>("RGVirtualMouse");
```
Comment (vontell) [inline src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ReplayDataPlaybackController.cs:293]: Very relieved we can do stuff like this with the new input system 😬

**Review by vontell — APPROVED** [review]
Looking good! Excited to try this out. Note that we had a question about states of monobehaviours as well, sent in the validations slack channel.

**Pull request merged**